File: api/Models/Entities/Notification.cs
using api.Models.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public enum NotificationType
{
    Generic,
    TaskAssigned,
    TaskDueSoon,
    TaskStatusChanged
}

public class Notification
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    [Required]
    public Guid RecipientUserId { get; set; }

    [ForeignKey(nameof(RecipientUserId))]
    public User? RecipientUser { get; set; }

    public Guid? RelatedTaskId { get; set; }

    [ForeignKey(nameof(RelatedTaskId))]
    public TaskItem? RelatedTask { get; set; }

    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(2000)]
    public string? Message { get; set; }

    [Required]
    public NotificationType Type { get; set; } = NotificationType.Generic;

    public bool IsRead { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ReadAt { get; set; }
}

File: client/src/app/types/Notification.ts
export const NotificationType = {
  Generic: 0,
  TaskAssigned: 1,
} as const;

export type UnreadCountResponse = {
  count: number;
};

export type NotificationType =
  (typeof NotificationType)[keyof typeof NotificationType];

export type NotificationDto = {
  id: string;
  recipientUserId: string;
  relatedTaskId?: string | null;
  title: string;
  message?: string | null;
  type: NotificationType;
  isRead: boolean;
  createdAt: string;
  readAt?: string | null;
};

export type CreateNotificationDto = {
  recipientUserId: string;
  title: string;
  message?: string | null;
  type?: NotificationType;
  relatedTaskId?: string | null;
};

export type PagedResult<T> = {
  items: T[];
  page: number;
  pageSize: number;
  total: number;
};

export type UseNotificationsParams = {
  page?: number;
  pageSize?: number;
};
